### PR TITLE
feat: `deno upgrade --rc`

### PR DESCRIPTION
**Pull request by bartlomieju**
Flag name open to bike-shedding.

### CODE REVIEW

File: cli/Cargo.toml
Comment (bartlomieju) [inline cli/Cargo.toml:5]: Revert for now

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:128]: Most likely remove; since canary and rc are gonna be pretty much the same and the only difference is that RC is just a designated canary release we might want to skip this particular check. Alternatively this could be an async fn and check against the designated canary to figure out if it's RC.

File: cli/Cargo.toml
Comment (bartlomieju) [inline cli/Cargo.toml:5]: Revert before merging

File: release-2-rc.txt
Comment (bartlomieju) [inline release-2-rc.txt:1]: Remove, add a GH actions script that appends to a file like this in GCP bucket

File: foo.js
Comment (bartlomieju) [inline foo.js:1]: Remove

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:621]: Add a comment why RC is considered canary

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:571]: Decide if we're gonna do RC blog post/and or link to notes

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:131]: Reveert

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:43]: Revert

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:101]: Remove, update a docstring

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:720]: Remove

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:807]: Add unit test

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:1038]: I opted to store this info in the `latest.txt` file so we don't have to hit remote server to display a banner. Works quite well 👍

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:1155]: Add a couple more tests

File: cli/tools/upgrade.rs
Comment (bartlomieju) [inline cli/tools/upgrade.rs:893]: Since this PR provides a generic solution to RCs it's probably best to rename it to `release-rc.txt`

**Comment by ry** [comment]
Should we also add `--lts` already as well?

**Comment by bartlomieju** [comment]
> Should we also add `--lts` already as well?

We can, but I think I want to try this mechanism first for a few days. I can do it once we're happy with it.

File: cli/args/flags.rs
Comment (littledivy) [inline cli/args/flags.rs:2885]:
```suggestion
          Arg::new("release-candidate")
```

other flags we have use `-` (like `dry-run`)

File: cli/tools/upgrade.rs
Comment (dsherret) [inline cli/tools/upgrade.rs:158]: This is realy hacky, but ok for the first pass.

**Review by dsherret — APPROVED** [review]
LGTM once the flag name is changed to use a dash.

**Pull request merged**
